network/src/p2p/builder.rs

```diff
@@ -73,19 +72,36 @@ impl TransportBuilder {
     }
 
     /// Builds libp2p transport
-    pub fn build<TTransport>(
+    pub fn build(
         self,
-        transport: TTransport,
-    ) -> (transport::Boxed<(PeerId, StreamMuxerBox)>, PeerId)
-    where
-        TTransport: Transport + Sized + Clone + Send + Sync + 'static,
-        <TTransport as Transport>::Output: AsyncRead + AsyncWrite + Send + Unpin + 'static,
-        TTransport::Dial: Send + 'static,
-        TTransport::Listener: Send + 'static,
-        TTransport::ListenerUpgrade: Send + 'static,
-        TTransport::Error: Send + Sync,
-    {
-        (
+    ) -> Result<(transport::Boxed<(PeerId, StreamMuxerBox)>, PeerId), std::io::Error> {
```

Comment:
would be good to create a type alias for this.

Reply:
Fixed